### PR TITLE
repo-tools: remove lint rule overrides

### PR DESCRIPTION
**Pull request by Rugvip**
## Hey, I just made a Pull Request!

Realized there was a lint rule override in `repo-tools` that will cause a bunch of breakages for people trying to use the package. @djamaile, @Sarabadu could you have a look? 👀, feel free to hijack the branch if you want to

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by Sarabadu** [comment]
@Rugvip yes working on it, good catch!!

File: yarn.lock
Comment (Sarabadu) [inline yarn.lock:8499]: update dependencies to unlock `@microsoft/tsdoc` version that was fixed

File: yarn.lock
Comment (Sarabadu) [inline yarn.lock:8501]: `@microsoft/api-extractor` is still depending on version `1.16.1` so need to mimic the version to not have tsc conflicts

File: packages/repo-tools/package.json
Comment (Sarabadu) [inline packages/repo-tools/package.json:38]: @microsoft/api-extractor

Comment (Sarabadu) [inline packages/repo-tools/package.json:38]: update version to unlock the `@microsoft/tsdoc`

File: packages/repo-tools/package.json
Comment (Rugvip) [inline packages/repo-tools/package.json:51]: It might make the most sense to make this a peer dependency, but not sure. I have a feeling 4.9 will break this repo for example, because we don't support it yet. Then again it might not be used for anything too important. I'd default to a peer dep with a pretty broad range, and then switch it up if that doesn't work well.

File: packages/repo-tools/package.json
Comment (Rugvip) [inline packages/repo-tools/package.json:49]: Perhaps make this a peer dep as well and only run it when available? Iirc the main purpose of running prettier is to have the API reports properly formatted out of the box to not have to run prettier manually afterwards. If prettier isn't installed that's not really a concern. It also makes it important to run it with the correct config.

File: packages/repo-tools/package.json
Comment (Rugvip) [reply packages/repo-tools/package.json:39]: Hmm, what do you mean? 1.16.1 of what?

Comment (Sarabadu) [reply packages/repo-tools/package.json:39]: nvm was 0.16.1 of `tsdoc-config`, but I'm taking a different approach here... I will keep all the versions as is to remove the linter and leave the update for a following PR

File: packages/repo-tools/package.json
Comment (Sarabadu) [reply packages/repo-tools/package.json]: yes, giving this a try

File: packages/repo-tools/package.json
Comment (Sarabadu) [inline packages/repo-tools/package.json:56]: not sure about it, but we need the exact same version as in the transitive dependecies from `@microsoft/api-extractor` and `@microsoft/api-documenter` if not we get the same issue with `material-ui` packages, since we are monkey  patching different versions

Comment (Sarabadu) [reply packages/repo-tools/package.json:56]: the alternative is to fix the version, thinking about what is better tbh

File: packages/repo-tools/src/commands/api-reports/api-extractor.ts
Comment (Sarabadu) [inline packages/repo-tools/src/commands/api-reports/api-extractor.ts:216]: all this typing maybe os more confusing than helping here but was the only way i find to have proper `prettier` typings.
Should test the behaviour without prettier installed, and if it casu any problem will remove all the type import

Comment (Sarabadu) [reply packages/repo-tools/src/commands/api-reports/api-extractor.ts:216]: tested and seems to work fine without any `prettier` installed, so keeping it

**Review by jhaals — CHANGES_REQUESTED** [review]
Thanks for looking at this @Sarabadu!  Let's wait for @Rugvip take a final look at the` peerDependencies` before merging this in. We also need a changeset to publish a new version of the package.

**Review by jhaals — APPROVED** [review]
Thanks! Approving to unblock, PTAL @Rugvip

File: packages/repo-tools/src/commands/api-reports/api-extractor.ts
Comment (Rugvip) [reply packages/repo-tools/src/commands/api-reports/api-extractor.ts:216]: You can `typeof import('prettier')` as well, but this works too :grin:

**Review by Rugvip — COMMENTED** [review]
Let's :shipit:

File: packages/repo-tools/src/commands/api-reports/api-extractor.ts
Comment (Sarabadu) [reply packages/repo-tools/src/commands/api-reports/api-extractor.ts:216]: ohh didn’t think about that, thanks!!

**Pull request merged**
